Protocol version 5

alexsporn opened the pull request:
# Description of change

Bump version to 0.11.0-alpha and introduce protocol version 5

valeriyr commented on crates/iota-protocol-config/src/lib.rs:
```diff
@@ -45,11 +46,6 @@ impl ProtocolVersion {
     #[cfg(not(msim))]
     const MAX_ALLOWED: Self = Self::MAX;
 
-    // We create 1 additional "fake" version in simulator builds so that we can
-    // test upgrades.
-    #[cfg(msim)]
-    pub const MAX_ALLOWED: Self = Self(MAX_PROTOCOL_VERSION + 1);
-
```
This constant is used in the [protocol version](https://github.com/iotaledger/iota/blob/380ce4d8223c369945dd2163b2790c8cf9d069f9/crates/iota-e2e-tests/tests/protocol_version_tests.rs#L98) simulation tests.
I think it should be reverted to contain one additional fake version as before.

alexsporn replied:
No, I think we just need to remove the `#[cfg(not(msim))]` from the normal MAX_ALLOWED.
@muXxer can you confirm?

miker83z replied:
Could just be 1 const like this: `pub const MAX_ALLOWED: Self = Self::MAX;` (check #5440).

valeriyr replied:
In this case, we also need to remove this part that changes the base transaction cost for the fake version:
https://github.com/iotaledger/iota/blob/187c27ae4b1e8a2874956e4d134004341046a0ea/crates/iota-protocol-config/src/lib.rs#L1193-L1201
p.s. In Sui the fake version still exists, it would be nice to check if it is really deprecated.

miker83z replied:
@valeriyr 
> p.s. In Sui the fake version still exists, it would be nice to check if it is really deprecated.

It seems not, i reverted that in #5440 -> [cebe617](https://github.com/iotaledger/iota/pull/5440/commits/cebe6171beffc6686a7bc5897c5e4cc67ccf6e86)

muXxer replied:
I would say we need to keep it for this one case were they want to inject a different base fee for the fake simulator version.

valeriyr committed:
fix(iota-framework): audit issue 58 extra mutable self in coin manager (#5447)

* fix(iota-framework): use immutable self in `coin_manager::additional_metadata`

* feat(iota-framework-snapshot): the snapshots update

* fix(iota-swarm-config): update the baseline

* fix(iota-framework): add `coin_manager::get_additional_metadata`

* feat(iota-framework-snapshot): the snapshots update

* fix(iota-swarm-config): update the baseline

alexsporn committed:
feat: bump version to 0.11.0-alpha with protocol version 5

revert: removal of fake `MAX_ALLOWED` for tests

cyberphysic4l committed:
feat(protocol): PCR-2 (#5479)

* PCR-2 base changes

* transaction-fuzzer changes

* comment construct_advance_epoch_pt_* functions

* iota-adapter-transactional-tests changes (#5283)

Co-authored-by: Mirko Zichichi <miker83z@proton.me>

* feat(protocol): PCR-2 iota-json-rpc* and iota-open-rpc (#5277)

* iota-json-rpc* and iota-open-rpc changes

* refactor(protocol): pcr2 more base changes (#5448)

* feat: implement conversions between v1 and v2 for system-state summary

* refactor(types): add non_exhaustive attribute to IotaSystemStateSummary

* refactor(core): use system-state summary v2

* doc(type): fix typo in system-state summary v2 doc

* refactor(json-rpc): support new system state summary (#5449)

* feat(json-rpc-api): support all versions of system-state summary

* refactor(json-rpc): support v2 of system-state summary

* chore(open-rpc): update specification

---------

Co-authored-by: Konstantinos Demartinos <konstantinos.demartinos@iota.org>

* feat(protocol): PCR-2 iota-rest-api (#5276)

* iota-rest-api changes

* refactor(rest-api): use v2 of system state summary (#5455)

---------

Co-authored-by: Konstantinos Demartinos <konstantinos.demartinos@iota.org>

* feat(protocol): PCR-2 iota-proxy (#5461)

* feat(protocol): PCR-2 iota-sdk (#5467)

* feat(protocol): PCR-2 iota-sdk

* more examples fix

* feat(protocol): PCR-2 iota-indexer (#5278)

* iota-indexer changes

* refactor(indexer): use new system state summary (#5477)

* refactor(indexer): add burned computation cost to schema

* refactor(indexer)!: use new system-state summary type

* Update crates/iota-indexer/src/handlers/checkpoint_handler.rs

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

---------

Co-authored-by: Konstantinos Demartinos <konstantinos.demartinos@iota.org>
Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

* feat(protocol): PCR-2 iota-bridge-cli/iota-tool (#5478)

* feat(protocol): PCR-2 iota-bridge-cli

* iota-tool

* refactor(types): derive serde and clone for IotaSystemStateSummary

* iota-analytics-indexer changes (#5282)

* feat(protocol): PCR-2 iota-graphql-rpc (#5280)

* iota-graphql-rpc changes

* feat(indexer): add computation_cost_burned getter

* refactor(graphql-rpc): use the new system-state summary

---------

Co-authored-by: Konstantinos Demartinos <konstantinos.demartinos@iota.org>

* refactor(transactional-test-runner): use new system-state summary type (#5501)

* fix(indexer): get system summary conditionally in index_epoch

* refactor(iota): use new system-state summary type (#5502)

* refactor(rosetta): use new system-state summary (#5507)

* refactor(graphql-e2e-tests: update baselines (#5281)

Co-authored-by: Konstantinos Demartinos <konstantinos.demartinos@iota.org>

* refactor(json-rpc-tests): use new system-state summary

* feat(protocol): PCR-2 iota-benchmark and iota-cluster-test (#5516)

* refactor(iota-benchmark): use new system-state summary type

* refactor(iota-cluster-test): use new system-state summary type

* fix: clippy error

* feat(protocol): PCR-2 iota-e2e-tests (#5520)

* refactor(iota-e2e-tests): use new system-state summary type

* fix: never trust the autocomplete...

* fix: typo

* fix: fix `IotaSystemStateV2` docstring

* fix: missing IotaSystemStateSummary in iota-e2e-tests simtests

* chores: Update system package build and snapshot

* chores: Update snapshot in iota-swarm-config

* Revert manifest.json in iota-framework-snapshot

* docs: Fix doc comments in iota_system_state_inner_v2.rs

---------

Co-authored-by: jkrvivian <jkrvivian@gmail.com>

* refactor(json-rpc,indexer): add get_latest_iota_system_state_v2 (#5586)

* refactor(json-rpc-api): add get_latest_iota_system_state_v2

* refactor(indexer): use get_latest_iota_system_state_v2

* refactor(json-rpc-api): add deprecated rust attribute

* Tokenomics documentation update (#5271)

* Initial commit tokenomics documentation update

* docs(tokenomics): first round of changes in tokenomics.mdx and gas-in-iota.mdx

* gas pricing updates

* Solved merge conflict

* Solved merge conflict

* staking changes

* docs(tokenomics): review iota-token and proof-of-stake

* docs(tokenomics): review validators-staking

* docs(tokenomics): review staking-unstaking

* docs(tokenomics): review gas-pricing

* docs(tokenomics): review gas-in-iota

* Tokenomics home page fix

* Remove redundant text related to token utilization

* Minor changes to tokenomics docs

* Update docs/content/about-iota/tokenomics/tokenomics.mdx

Co-authored-by: Lucas Tortora <85233773+lucas-tortora@users.noreply.github.com>

* Style and naming changes in gas pricing file

* fix inflation rate value

* Update docs/content/_snippets/staking-pool-reqs.mdx

Co-authored-by: Lucas Tortora <85233773+lucas-tortora@users.noreply.github.com>

* Update docs/content/about-iota/tokenomics/gas-pricing.mdx

Co-authored-by: Lucas Tortora <85233773+lucas-tortora@users.noreply.github.com>

* tweaks to tokenomics homepage

* Update tokenomics image

* update tokenomics image

---------

Co-authored-by: Roman Overko <roman.overko@iota.org>
Co-authored-by: Andrew <i.am.andrew.cullen@gmail.com>
Co-authored-by: Can <canumutileri@gmail.com>
Co-authored-by: Lucas Tortora <85233773+lucas-tortora@users.noreply.github.com>
Co-authored-by: Lucas Tortora <lucas.tortora@iota.org>
Co-authored-by: Andrew Cullen <45826600+cyberphysic4l@users.noreply.github.com>

* refactor(iota-sdk): version get_latest_iota_system_state (#5587)

* refactor(json-rpc-api): add get_latest_iota_system_state_v2

* refactor(indexer): use get_latest_iota_system_state_v2

* refactor(iota-sdk): version get_latest_iota_system_state

* refactor(iota-sdk): use get_latest_iota_system_state_v2 if available in the rpc_methods

* refactor(iota-sdk): cache iota_system_state_v2_support

* fix: use get_latest_iota_system_state_v2 in tests

* fix: update remaining tests to get_latest_iota_system_state_v2

---------

Co-authored-by: Thoralf Müller <thoralf.mue@gmail.com>
Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

* refactor(simulacrum): use ChangeEpochV2

* chore(json-rpc-api,open-rpc): correct supported protocol version in docs

* feat(ts-sdk): update TS SDK after PCR2 changes (#5637)

* feat(update-tssdk-types): Run update-open-rpc-client-types

* feat(update-tssdk-types): Support versioned system state. Sdk build passing. Updates version.ts.

* feat(update-tssdk-types): Bump graphql schema version (2025.2) and run update-graphql-schemas

* feat(update-tssdk-types): Run graphql-transport codegen

* feat(update-tssdk-types): Fix IotaSystemStateSummary type mismatch in graphql-transport

* feat(update-tssdk-types): Support getLatestIotaSystemStateV2 in TS client and adjust

* feat(update-tssdk-types): Fix turbo build & lint for apps

* feat(ts-sdk): Add changeset

* Update sdk/typescript/src/client/client.ts comment to align with openrpc.json

Co-authored-by: Begoña Álvarez de la Cruz <balvarez@boxfish.studio>

* feat(ts-sdk): dprint

* feat(ts-sdk): dprint

---------

Co-authored-by: Begoña Álvarez de la Cruz <balvarez@boxfish.studio>

---------

Co-authored-by: Alexander Sporn <github@alexsporn.de>
Co-authored-by: Mirko Zichichi <miker83z@proton.me>
Co-authored-by: Konstantinos Demartinos <konstantinos.demartinos@iota.org>
Co-authored-by: muXxer <mux3r@web.de>
Co-authored-by: Thibault Martinez <thibault@iota.org>
Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>
Co-authored-by: jkrvivian <jkrvivian@gmail.com>
Co-authored-by: Luigi Vigneri <luigivigneri89@gmail.com>
Co-authored-by: Roman Overko <roman.overko@iota.org>
Co-authored-by: Can <canumutileri@gmail.com>
Co-authored-by: Lucas Tortora <85233773+lucas-tortora@users.noreply.github.com>
Co-authored-by: Lucas Tortora <lucas.tortora@iota.org>
Co-authored-by: Thoralf Müller <thoralf.mue@gmail.com>
Co-authored-by: Mario <mario.sarcevic@iota.org>
Co-authored-by: Begoña Álvarez de la Cruz <balvarez@boxfish.studio>
